### ISSUE TITLE
Opened in readonly mode for no reason

### ISSUE THREAD

**Issue by georgit** [keepassxreboot/keepassxc#1937]
When I opened KeePassXC on Ubuntu 16.04, brought up the key login page for the db I always use. After entering that, it listed my entries but told me it was in readonly mode. I thought this was odd because I opened it same as always from the Unity launcher. It let me add an entry and remove an entry I realized I had added twice but when I tried to quit, it asked me to save. Then it couldn't save the db file due to permission denied. I ended up closing without saving. Then I opened it again and it told me it was readonly again. I closed without making changes this time and then opened a third time. This time it made me choose the db file to open which has never happened before. But once I picked it, the entries were displayed in read/write mode so I could make changes. I have been using it for 2+ months now and this has never happened before.

## Expected Behavior
If I open KeePassXC and it always defaults to read/write mode, I would expect that to be consistent. It would also be nice if in readonly mode, it didn't let you add/remove entries in the first place, but this might be harder. I'm just not sure why it opened in readonly mode twice in a row despite the same startup steps taken.

## Current Behavior
KeePassXC seems to arbitrarily choose what mode it opens in.

## Context
This is a personal account so it's not linked to any network. Just a keypass database file on my system.

## Debug Info
KeePassXC - Version 2.3.3
Revision: 0a155d8
Distribution: Snap

Libraries:
- Qt 5.7.1
- libgcrypt 1.7.6-beta

Operating system: Ubuntu Core 16
CPU architecture: x86_64
Kernel: linux 4.13.0-41-generic

Enabled extensions:
- Auto-Type
- Browser Integration
- Legacy Browser Integration (KeePassHTTP)
- SSH Agent
- YubiKey

Libraries:
- LIBS

Operating system: OS
CPU architecture: ARCH
Kernel: KERNEL

Enabled extensions:
- EXTENSIONS

**Comment by droidmonkey** [comment]
I'll take a look, are you opening the database from your home directory?

**Comment by droidmonkey** [comment]
This is an issue when snaps update. I cannot do anything about it, even if you open from the ~/snap/keepassxc/current symlink folder it still causes read-only opening after a version bump. I DO NOT recommend storing your kdbx in the snap folder.

**Comment by georgit** [comment]
I see. Thank you for looking into it! I'll move the kdbx out of the snap folder. It hasn't happened since then. Maybe the default kdbx store location should be changed (I think this was just the default suggested) but this explains everything.

**Comment by droidmonkey** [comment]
The default is the user's home directory, but the snap containment fakes that to be the snap directory. I could look at making an exception for snaps to go to the user home.

**Comment by droidmonkey** [comment]
I am going to make a new issue for the snap folder, I just found that if you uninstall keepassxc snap it will DELETE all files in the snap folder without any warning. That means you could lose your database.

**Comment by PenelopeFudd** [comment]
This is November 2019, and my ~/snap/keepassxc/625/Passwords.kdbx database suddenly became readonly.  

To get it to work again, I copied that file to my home directory, ran 'snap remove keepassxc; snap install keepassxc', started keepassxc, created a new database with all defaults, quit the program, then copied the database back into the ~/snap/keepassxc/625 directory.

Upon starting keepassxc after that, it was read-write again.

When I tried accessing the database in my home directory, it was still read-only.
